sphinx/source/developers/consensus.rst

```diff
@@ -0,0 +1,41 @@
+Consensus Protocols
+===================
+
+In order to guarantee system reliability changes to the key-value store must be agreed by a number of nodes before being applied. The number of nodes that need to agree on a change before it is applied depends on the consensus protocol being run.
+
+The default consensus protocol for CCF is a version of Raft which provides Crash Fault Tolerance.
+
+We are currently working on providing PBFT as an additional consensus option. PBFT provides Byzantine Fault Tolerance.
+
+.. warning:: Currently CCF with PBFT is in development and should not be used in a production environment.
+
+Raft Consensus Protocol
+-----------------------
+
+We have implemented a version of Raft that provides Crash Fault Tolerance. This means that if we can tolerate up to ``f`` nodes crashing (or being un-responsive) then we need ``2*f + 1`` nodes in total to guarantee liveness. The majority of nodes that need to agree on a change is then ``f + 1``.
```

Comment:
I don't like "The majority of nodes that need to agree on a change", here or for PBFT - I think it's worth clarifying that the form of agreement is very different. Maybe something like:

`Raft will only commit a change once it has been shared with f+1 nodes`
`PBFT will only commit a change once 2f+1 nodes have confirmed the same execution result`